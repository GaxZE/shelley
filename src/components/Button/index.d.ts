/**
 * Button props extending those of a regular button.
 */
export interface ButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  /** Type of the button default, primary..*/
  color?: string;
  /** Optional ref. */
  ref?: React.Ref<HTMLButtonElement>;
  /** Button sizes: xs, sm, md, lg */
  size?: string;
  /** Button variant. */
  variant?: string;
  /** Extra text that can be used to render a tooltip on hover/focus. */
  tip?: string;
}
