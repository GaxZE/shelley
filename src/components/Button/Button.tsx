import React from "react";
import type { Accent, AlignPos, Volume, Variant } from "../types";
import type { MergeElementProps } from "../utils";
/* = Style API. */
import { st, classes } from "./button.st.css";

/**
 * Props
 */

export interface ButtonBaseProps {
  /** Define an Icon node, postion via #iconPos. */
  icon?: React.ReactNode;
  /** The position of the icon relative to the label. */
  iconPos?: AlignPos;
  /** Tone index, defines the color palette. */
  tone?: Accent;
  /** Variant index, defines the 'look'. */
  variant?: Variant;
  /** Defines how 'loud' the Button should be in term of it's size. */
  vol?: Volume;
  /** Applies width of 100%. */
  fullWidth?: boolean;
}

export type ButtonProps<P extends React.ElementType = "button"> = {
  /** Custom element to render such as an anchor "a" or a router "Link" component. */
  as?: P;
} & MergeElementProps<P, ButtonBaseProps>;

function ButtonBase<T extends React.ElementType = "button">(
  {
    children,
    className: classNameProp,
    as,
    icon,
    iconPos = "end",
    fullWidth = false,
    tone = 1,
    variant = 1,
    vol = 3,
    ...rest
  }: ButtonProps<T>,
  ref: React.Ref<T>
) {
  const className = st(
    classes.root,
    {
      iconPos: icon ? iconPos : undefined,
      fullWidth,
      tone,
      variant,
      vol,
    },
    classNameProp
  );
  const internal = (
    <>
      {icon && (
        <>
          {icon}
          {children && <span className={classes.divider}></span>}
        </>
      )}
      {children && <span className={classes.inner}>{children}</span>}
    </>
  );
  return React.createElement(
    as || "button",
    {
      ref,
      ...rest,
      className,
    },
    internal
  );
}

const Button = React.forwardRef(ButtonBase) as typeof ButtonBase;

export default Button;
