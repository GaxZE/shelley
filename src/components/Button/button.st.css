/* button.st.css */
.button {}

.inner {}

.xs {}

.sm {}

.md {}

.lg {}

.xl {}

/* How to theme, better to use CSS vars? */
.default {}

.primary {}

.secondary {}

.fill {}

.hollow {}

/*State selecytor*/
.disabled {}
