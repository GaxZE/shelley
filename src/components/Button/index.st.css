.root {
  background: pink
}
Button {
  background: yellow
}

Button.test {
  background: pink
}

:import {
  -st-from: "@bc-digital/bronte/dist/components/Button/button.st.css";
  -st-default: Button;
}
