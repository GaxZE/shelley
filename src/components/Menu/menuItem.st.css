/* menuItem.st.css CSS API */

.root {
  display: block;
  user-select: none;
  cursor: pointer;
  color: inherit;
  font: inherit;
  text-decoration: initial;
}

/* pseudo pseudo selector */
.root[data-selected] {}

.root[disabled] {
  opacity: 0.5;
  cursor: not-allowed;
}
