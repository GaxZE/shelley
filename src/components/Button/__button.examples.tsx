/* buttonExamples.tsx */
import React from "react";
/* Example Component */
import Button from "./Button";
/* Supporting Components */
import Icon from "../../components/Icon/Icon";
import PropsDemo from "../../components_site/PropsDemo/PropsDemo";
import CodeSample from "../../components_site/CodeSample/CodeSample";
import { classes as grid } from "../../styles/default/grid.st.css";
import { classes as spacing } from "../../styles/default/spacing.st.css";
export const meta = {
  name: "Button"
};
import { H2, P } from "../../components/Text/Text";

export const QuickRef = () => (
  <CodeSample className={grid.mid}>{`import Button from "@actionishope/shelley";
\n<Button>Climate fight club</Button>
/* One rule: Fight climate change, not each other. */`}</CodeSample>
);

// Props Demo - Examples below.
export const ComponentDemo = () => {
  const [labelDemoProps, setLabelDemoProps]: any = React.useState([
    {
      name: "children",
      label: "children",
      type: "text",
      value: "Climate fight club"
    },
    {
      name: "tone",
      label: "tone",
      type: "number",
      min: 0,
      max: 6,
      value: 1
    },
    {
      name: "variant",
      label: "variant",
      type: "number",
      min: 0,
      max: 6,
      value: 3
    },
    {
      name: "volume",
      label: "volume",
      type: "number",
      min: 0,
      max: 6,
      value: 3
    },
    {
      name: "icon",
      label: "icon",
      type: "select",
      value: "No",
      options: ["Yes", "No"]
    },
    {
      name: "iconPos",
      label: "iconPos",
      type: "select",
      value: "end",
      options: ["top", "end", "bottom", "start"]
    }
  ]);
  return (
    <PropsDemo
      demoProps={labelDemoProps}
      componentName="Button"
      setDemoProps={setLabelDemoProps}
      tsExtends="HTMLButtonElement"
      renderExample={
        <CodeSample>{`import { Button, Icon } from "@actionishope/shelley";\n\n<Button ${
          labelDemoProps[1].value > 0
            ? `\n  tone={${labelDemoProps[1].value}}`
            : `\n  tone={false}`
        } ${
          labelDemoProps[2].value > 0
            ? `\n  variant={${labelDemoProps[2].value}}`
            : `\n  variant={false}`
        } ${
          labelDemoProps[3].value > 0
            ? `\n  vol={${labelDemoProps[3].value}}`
            : `\n  vol={false}`
        } ${
          labelDemoProps[4].value === "Yes"
            ? `\n  icon={<Icon><path d="M14 7h-5v-5h-2v5h-5v2h5v5h2v-5h5v-2z"></path></Icon>}`
            : ``
        } ${
          labelDemoProps[4].value === "Yes"
            ? `\n  iconPos={"${labelDemoProps[5].value}"}`
            : ``
        }  \n>\n ${labelDemoProps[0].value}\n</Button>`}</CodeSample>
      }
    >
      <Button
        tone={labelDemoProps[1].value === 0 ? false : labelDemoProps[1].value}
        icon={
          labelDemoProps[4].value === "No" ? (
            false
          ) : (
            <Icon>
              <path d="M14 7h-5v-5h-2v5h-5v2h5v5h2v-5h5v-2z"></path>
            </Icon>
          )
        }
        iconPos={labelDemoProps[5].value}
        variant={
          labelDemoProps[2].value === 0 ? false : labelDemoProps[2].value
        }
        vol={labelDemoProps[3].value === 0 ? false : labelDemoProps[3].value}
      >
        {labelDemoProps[0].value}
      </Button>
    </PropsDemo>
  );
};

export const ComponentHTML = () => (
  <CodeSample>{`<button class="button-root button-tone1 button-variant1 button-vol2">
  <span class="button-inner">Earth Guardians</span>
</button>`}</CodeSample>
);

export const ComponentCSS = () => (
  <CodeSample fixedHeight>{`/** button.st.css */

/* Root */
Button {}

/* Parts */

/* 'inner' button, use for vertical spacing to give a potential Icon more room */
Button::inner {}

/* The divider which is rendered with an icon only. */
Button::divider {}

/* A basic CSS tooltip. */
Button::tip {}


/* Accent colours */
Button.tone1 {}
Button.tone2 {}
Button.tone3 {}
Button.tone4 {}
Button.tone5 {}
Button.tone6 {}

/* Volumes */
Button.vol1 {}
Button.vol2 {}
Button.vol3 {}
Button.vol4 {}
Button.vol5 {}
Button.vol6 {}

/* Variants */
Button.variant1 {}
Button.variant2 {}
Button.variant3 {}
Button.variant4 {}
Button.variant5 {}
Button.variant6 {}`}</CodeSample>
);

export const Example1 = () => (
  <>
    <H2 vol={2} className={spacing.mb2} uppercase>
      Anchor link Buttons
    </H2>

    <P className={spacing.mb2}>
      To use <code>Button</code> styles in conjuction with a link you can use
      the <code>as</code> to provide a cutom <code>Link</code> component such as{" "}
      <code>gatsby-link</code> or just a plain old <code>a</code>.
    </P>

    <div className={grid.mid}>
      <Button
        variant={3}
        as={"a"}
        to="https://www.w3schools.com/html/html_links.asp"
      >
        Basic anchor link
      </Button>
    </div>
    <CodeSample className={grid.mid}>
      {`import Link from "gatsby-link";
/* use any Link component that supports a 'to' prop. */

<Button variant={3} as={Link} to="/path-to-resource">
  Custom Link component
</Button>

or

<Button variant={3} as={"a"} to="/path-to-resource">
  Anchor link Button
</Button>`}
    </CodeSample>
  </>
);
