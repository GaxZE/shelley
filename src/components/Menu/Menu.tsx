/**
 * Menu.tsx
 *
 * This file wraps the awesome Reach UI 'MenuButton' collection.
 * https://reach.tech/menu-button
 *
 * Please do checkout the library, it is very good, it has been
 * very useful in developing this lib.
 *
 */

import React from "react";

import classnames from "classnames";
import {
  Menu as ReachMenu,
  MenuButton as ReachMenuButton,
  MenuList as ReachMenuList,
  MenuItems as ReachMenuItems,
  MenuItem as ReachMenuItem,
  MenuPopover as ReachMenuPopover,
  MenuLink as ReachMenuLink,
  MenuProps as ReachMenuProps,
  // MenuButtonProps as ReactMenuButtonProps,
  MenuListProps as ReachMenuListProps,
  MenuItemsProps as ReachMenuItemsProps,
  MenuItemProps as ReachMenuItemProps,
  MenuPopoverProps as ReachMenuPopoverProps,
  MenuLinkProps as ReachMenuLinkProps
} from "@reach/menu-button";
import "./styles.css";

import Button, { ButtonProps } from "../Button/Button";
import { st as stList, classes as list } from "./menuList.st.css";
import { st as stItem, classes as item } from "./menuItem.st.css";

export type MenuProps = ReachMenuProps;
export type MenuButtonProps = ButtonProps;
export interface MenuListProps extends ReachMenuListProps {
  as?: "ul" | "div";
}
export type MenuItemsProps = ReachMenuItemsProps;
export interface MenuItemProps extends ReachMenuItemProps {
  className?: string;
  disabled?: boolean;
  as?: "li" | "div";
}
export type MenuPopoverProps = ReachMenuPopoverProps;
export type MenuLinkProps = ReachMenuLinkProps;

/**
 * Reach UI Menu - Straight wrap and go of the main menu.
 */
export const Menu = ({ children, ...rest }: MenuProps) => (
  <ReachMenu {...rest}>{children}</ReachMenu>
);

/**
 * Reach UI MenuButton - Churned out as Shelley button which provides styles.
 */
export const MenuButton = React.forwardRef(
  (
    { children, ...rest }: MenuButtonProps,
    ref?: React.Ref<HTMLButtonElement>
  ) => (
    <Button as={ReachMenuButton} ref={ref} {...rest}>
      {children}
    </Button>
  )
);
MenuButton.displayName = "MenuButton";

/**
 * Reach UI MenuList - Decorated with stylable stylesheet.
 */
export const MenuList = React.forwardRef(
  (
    { className, children, ...rest }: MenuListProps,
    ref?: React.Ref<HTMLDivElement>
  ) => (
    <ReachMenuList
      className={stList(classnames(list.root, className))}
      ref={ref}
      {...rest}
    >
      {children}
    </ReachMenuList>
  )
);
MenuList.displayName = "MenuList";

/**
 * Reach UI MenuItems - see Reach docs, ccustom uses cases so no class provided.
 */
export const MenuItems = React.forwardRef(
  (
    { children, ...rest }: MenuItemsProps,
    ref?: React.Ref<HTMLUListElement | HTMLDivElement>
  ) => (
    <ReachMenuItems ref={ref} {...rest}>
      {children}
    </ReachMenuItems>
  )
);
MenuItems.displayName = "MenuItems";

/**
 * Reach UI MenuItem - Decorated with stylable stylesheet.
 */
export const MenuItem = React.forwardRef(
  (
    { className, children, as = "div", ...rest }: MenuItemProps,
    ref?: React.Ref<HTMLLIElement | HTMLDivElement>
  ) => (
    <ReachMenuItem
      className={stItem(classnames(item.root, className))}
      ref={ref}
      {...rest}
    >
      {children}
    </ReachMenuItem>
  )
);
MenuItem.displayName = "MenuItem";

/**
 * Reach UI MenuPopover - see Reach docs for custom use cases, straight wrap.
 */
export const MenuPopover = React.forwardRef(
  (
    { children, ...rest }: MenuPopoverProps,
    ref?: React.Ref<HTMLDivElement>
  ) => (
    <ReachMenuPopover ref={ref} {...rest}>
      {children}
    </ReachMenuPopover>
  )
);
MenuPopover.displayName = "MenuPopover";
