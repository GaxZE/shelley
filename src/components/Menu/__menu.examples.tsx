/* menuExamples.tsx */
import React from "react";
/* Example Component */
import {
  Menu,
  MenuList,
  MenuButton,
  MenuItem,
  MenuItems,
  MenuPopover
} from "./Menu";
/* Supporting Components */
import Icon from "../Icon/Icon";
import Button from "../Button/Button";
import ButtonGroup from "../ButtonGroup/ButtonGroup";
import Text, { H2 } from "../Text/Text";
import PropsDemo from "../../components_site/PropsDemo/PropsDemo";
import CodeSample from "../../components_site/CodeSample/CodeSample";
import { classes as grid } from "../../styles/default/grid.st.css";
import { classes as spacing } from "../../styles/default/spacing.st.css";

export const meta = {
  name: "Menu"
};

export const QuickRef = () => (
  <CodeSample className={grid.mid}>{`import {
    Menu,
    MenuList,
    MenuButton
  } from "@action-is-hope/shelley";
\n
<Menu>
  <MenuButton
    variant={3}
  >
    Actions â–¾
  </MenuButton>
  <MenuList>
    <MenuItem onSelect={() => alert("1")}>Option One</MenuItem>
    <MenuItem onSelect={() => alert("2")}>Option Two</MenuItem>
    <MenuItem onSelect={() => alert("3")}>Option Three</MenuItem>
  </MenuList>
</Menu>`}</CodeSample>
);

// Props Demo - Examples below.
export const ComponentDemo = () => {
  const [labelDemoProps, setLabelDemoProps]: any = React.useState([
    {
      name: "children",
      label: "children",
      type: "text",
      value: "Earth Solutionists required"
    },
    {
      name: "tone",
      label: "tone",
      type: "number",
      min: 0,
      max: 6,
      value: 1
    },
    {
      name: "variant",
      label: "variant",
      type: "number",
      min: 0,
      max: 6,
      value: 1
    },
    {
      name: "volume",
      label: "volume",
      type: "number",
      min: 0,
      max: 6,
      value: 2
    },
    {
      name: "icon",
      label: "icon",
      type: "select",
      value: "No",
      options: ["Yes", "No"]
    },
    {
      name: "tip",
      label: "tip",
      type: "text",
      value: "@todo - unsure on this..."
    }
  ]);
  return (
    <PropsDemo
      demoProps={labelDemoProps}
      componentName="Button"
      setDemoProps={setLabelDemoProps}
      tsExtends="HTMLButtonElement"
      renderExample={
        <CodeSample>{`import { Button, Icon } from "@action-is-hope/shelley";\n\n<Button ${
          labelDemoProps[1].value > 0
            ? `\n  tone={${labelDemoProps[1].value}}`
            : `\n  tone={false}`
        } ${
          labelDemoProps[2].value > 0
            ? `\n  variant={${labelDemoProps[2].value}}`
            : `\n  variant={false}`
        } ${
          labelDemoProps[3].value > 0
            ? `\n  vol={${labelDemoProps[3].value}}`
            : `\n  vol={false}`
        } ${
          labelDemoProps[4].value === "Yes"
            ? `\n  icon={<Icon><path d="M14 7h-5v-5h-2v5h-5v2h5v5h2v-5h5v-2z"></path></Icon>}`
            : ``
        }  \n>\n ${labelDemoProps[0].value}\n</Button>`}</CodeSample>
      }
    >
      {/* <MenuButton
        tone={labelDemoProps[1].value === 0 ? false : labelDemoProps[1].value}
        icon={
          labelDemoProps[4].value === "No" ? (
            false
          ) : (
            <Icon alt={labelDemoProps[1].value}>
              <path d="M14 7h-5v-5h-2v5h-5v2h5v5h2v-5h5v-2z"></path>
            </Icon>
          )
        }
        variant={
          labelDemoProps[2].value === 0 ? false : labelDemoProps[2].value
        }
        vol={labelDemoProps[3].value === 0 ? false : labelDemoProps[3].value}
      >
        {labelDemoProps[0].value}
      </MenuButton> */}
      {/* <Button
        tone={labelDemoProps[1].value === 0 ? false : labelDemoProps[1].value}
        icon={
          <Icon>
            <path d="M14 7h-5v-5h-2v5h-5v2h5v5h2v-5h5v-2z"></path>
          </Icon>
        }
        variant={
          labelDemoProps[2].value === 0 ? false : labelDemoProps[2].value
        }
        vol={labelDemoProps[3].value === 0 ? false : labelDemoProps[3].value}
      >
        &nbsp;
      </Button> */}
    </PropsDemo>
  );
};

export const ComponentHTML = () => (
  <CodeSample>{`<button class="button-root button-tone1 button-variant1 button-vol2">
  <span class="button-inner">Earth Guardians</span>
</button>`}</CodeSample>
);

export const ComponentCSS = () => (
  <CodeSample fixedHeight>{`/** button.st.css */

/* Root */
Button {}

/* Parts */

/* 'inner' button, use for vertical spacing to give a potential Icon more room */
Button::inner {}

/* The divider which is rendered with an icon only. */
Button::divider {}

/* A basic CSS tooltip. */
Button::tip {}


/* Accent colours */
Button.tone1 {}
Button.tone2 {}
Button.tone3 {}
Button.tone4 {}
Button.tone5 {}
Button.tone6 {}

/* Volumes */
Button.vol1 {}
Button.vol2 {}
Button.vol3 {}
Button.vol4 {}
Button.vol5 {}
Button.vol6 {}

/* Variants */
Button.variant1 {}
Button.variant2 {}
Button.variant3 {}
Button.variant4 {}
Button.variant5 {}
Button.variant6 {}`}</CodeSample>
);

export const Example1 = () => (
  <>
    <H2 vol={2} className={spacing.mb2} uppercase>
      Icon/Text Menu Button example
    </H2>
    <div className={grid.mid}>
      <Menu>
        <MenuButton
          vol={3}
          tone={2}
          variant={3}
          iconPos="start"
          icon={
            <Icon alt="Block settings">
              <path d="M14 7h-5v-5h-2v5h-5v2h5v5h2v-5h5v-2z"></path>
            </Icon>
          }
        >
          Add Content
        </MenuButton>
        <MenuList>
          <MenuItem onSelect={() => alert("selected!")}>Page</MenuItem>
          <MenuItem onSelect={() => alert("selected!")}>Blog post</MenuItem>
          <MenuItem onSelect={() => alert("selected!")}>Event</MenuItem>
        </MenuList>
      </Menu>
    </div>
    <CodeSample className={grid.mid}>
      {`<Menu>
  <MenuButton
    vol={3}
    tone={2}
    variant={3}
    iconPos="start"
    icon={
      <Icon alt="Block settings">
        <path d="M14 7h-5v-5h-2v5h-5v2h5v5h2v-5h5v-2z"></path>
      </Icon>
    }
  >
    Add Content
  </MenuButton>
  ... see quick reference
</Menu>`}
    </CodeSample>
  </>
);

export const Example2 = () => (
  <>
    <H2 vol={2} className={spacing.mb2} uppercase>
      Button Group example
    </H2>
    <div className={grid.mid}>
      <Menu>
        <ButtonGroup vol={4} tone={2} variant={3}>
          <Button fullWidth>Publish</Button>
          <MenuButton
            icon={
              <Icon alt="Block settings">
                <path d="M13 4v2l-5 5-5-5v-2l5 5z"></path>
              </Icon>
            }
            onMouseDown={() => console.log("hi2")}
          />
        </ButtonGroup>
        <MenuList>
          <Text as="label" uppercase vol={1}>
            Change status
          </Text>
          <MenuItem disabled onSelect={() => alert("selected!")}>
            Publish
          </MenuItem>
          <MenuItem onSelect={() => alert("selected!")}>Archive</MenuItem>
          <MenuItem onSelect={() => alert("selected!")}>Delete</MenuItem>
        </MenuList>
      </Menu>
    </div>
    <CodeSample className={grid.mid}>
      {`<Menu>
  <ButtonGroup vol={4} tone={2} variant={3}>
    <Button fullWidth>Publish</Button>
    <MenuButton
      icon={
        <Icon alt="">
          <path d="M13 4v2l-5 5-5-5v-2l5 5z"></path>
        </Icon>
      }
      onMouseDown={() => console.log("hi2")}
    />
  </ButtonGroup>
  <MenuList>
    <Text as="label" uppercase vol={1}>
      Change status
    </Text>
    <MenuItem disabed onSelect={() => alert("selected!")}>Publish</MenuItem>
    <MenuItem onSelect={() => alert("selected!")}>Archive</MenuItem>
    <MenuItem onSelect={() => alert("selected!")}>Delete</MenuItem>
  </MenuList>
</Menu>`}
    </CodeSample>
  </>
);

// Add some tests to check refs and tings are spread properly.
// const inputEl = React.useRef();
// const test = React.createRef<HTMLButtonElement>();
// const onButtonClick = () => {
//   const node = test.current;
//   // `current` points to the mounted text input element
//   node && node.focus();
// };
