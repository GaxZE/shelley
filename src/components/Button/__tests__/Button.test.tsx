import React from "react";
import Button from "../Button";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import Icon from "../../Icon/Icon";
// import { Link } from "react-router-dom";

import { MenuButton as ReachMenuButton } from "@reach/menu-button";
import { Menu } from "../../Menu/Menu";

const buttonRef = React.createRef<HTMLButtonElement>();
const BigButtsString = "I love big buttons and I cannot lie...";
const iconPath = <path d="M14 7h-5v-5h-2v5h-5v2h5v5h2v-5h5v-2z"></path>;

describe("Button", () => {
  it("renders as a basic button with custom #className", () => {
    const tree = renderer
      .create(
        <Button
          data-testid="button-data-testid"
          className="for-which-nobody-can-deny"
          onClick={() => console.log("onClick")}
          onMouseEnter={() => console.log("onMouseEnter")}
          ref={buttonRef}
        >
          {BigButtsString}
        </Button>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders as disabled", () => {
    const DisabledButton = () => (
      <Button data-testid="button-data-testid" ref={buttonRef} disabled>
        {BigButtsString}
      </Button>
    );
    const { getByTestId } = render(<DisabledButton />);
    expect(getByTestId("button-data-testid")).toHaveAttribute("disabled");
  });

  it("renders with stylable state prop classes, #tone, #variant, #vol and #fullwidth", () => {
    const tree = renderer
      .create(
        <Button
          data-testid="button-data-testid"
          onClick={() => console.log("onClick")}
          onMouseEnter={() => console.log("onMouseEnter")}
          ref={buttonRef}
          tone={6}
          variant={6}
          vol={6}
          fullWidth
        >
          {BigButtsString}
        </Button>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders as a basic anchor link via #href prop", () => {
    const tree = renderer
      .create(
        <Button
          data-testid="button-data-testid"
          onClick={() => console.log("onClick")}
          onMouseEnter={() => console.log("onMouseEnter")}
          href="http://shelley.earth"
          ref={buttonRef}
        >
          {BigButtsString}
        </Button>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it("renders as third party anchor provided via #as and #t props", () => {
  //   const tree = renderer
  //     .create(
  //       <Button
  //         as={Link}
  //         data-testid="button-data-testid"
  //         to="internal-link"
  //         ref={buttonRef}
  //       >
  //         {BigButtsString}
  //       </Button>
  //     )
  //     .toJSON();
  //   expect(tree).toMatchSnapshot();
  // });

  it("includes #icon via prop positioned via #iconPos divider and children", () => {
    const tree = renderer
      .create(
        <Button
          data-testid="button-data-testid"
          ref={buttonRef}
          icon={<Icon>{iconPath}</Icon>}
          iconPos="start"
        >
          {BigButtsString}
        </Button>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("includes #icon, no children, visually hidden label text provided via Icon alt prop", () => {
    const tree = renderer
      .create(
        <Button
          data-testid="button-data-testid"
          ref={buttonRef}
          variant={4}
          icon={<Icon alt={BigButtsString}>{iconPath}</Icon>}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  // @todo fix jestconfig to map ONLY CSS files to identity-obj-proxy to support Reach CSS check.
  it("supports Reach Menu Button via #as", () => {
    const tree = renderer
      .create(
        <Menu>
          <Button
            as={ReachMenuButton}
            data-testid="button-data-testid"
            ref={buttonRef}
          >
            {BigButtsString}
          </Button>
        </Menu>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
