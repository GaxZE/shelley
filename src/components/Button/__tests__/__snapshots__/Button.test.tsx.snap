// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button includes #icon via prop positioned via #iconPos divider and children 1`] = `
<button
  className="Button2465625684__root Button2465625684---iconPos-5-start Button2465625684---tone-1-1 Button2465625684---variant-1-1 Button2465625684---vol-1-3"
  data-testid="button-data-testid"
>
  <svg
    aria-hidden={true}
    className="Icon2050503392__root"
    focusable="false"
    viewBox="0 0 16 16"
  >
    <path
      d="M14 7h-5v-5h-2v5h-5v2h5v5h2v-5h5v-2z"
    />
  </svg>
  <span
    className="Button2465625684__divider"
  />
  <span
    className="Button2465625684__inner"
  >
    I love big buttons and I cannot lie...
  </span>
</button>
`;

exports[`Button includes #icon, no children, visually hidden label text provided via Icon alt prop 1`] = `
<button
  className="Button2465625684__root Button2465625684---iconPos-3-end Button2465625684---tone-1-1 Button2465625684---variant-1-4 Button2465625684---vol-1-3"
  data-testid="button-data-testid"
>
  <svg
    aria-hidden={true}
    className="Icon2050503392__root"
    focusable="false"
    viewBox="0 0 16 16"
  >
    <path
      d="M14 7h-5v-5h-2v5h-5v2h5v5h2v-5h5v-2z"
    />
  </svg>
  <span
    className="visuallyHidden930791739__root"
  >
    I love big buttons and I cannot lie...
  </span>
</button>
`;

exports[`Button renders as a basic anchor link via #href prop 1`] = `
<a
  className="Button2465625684__root Button2465625684---tone-1-1 Button2465625684---variant-1-1 Button2465625684---vol-1-3"
  data-testid="button-data-testid"
  href="http://shelley.earth"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <span
    className="Button2465625684__inner"
  >
    I love big buttons and I cannot lie...
  </span>
</a>
`;

exports[`Button renders as a basic button with custom #className 1`] = `
<button
  className="Button2465625684__root Button2465625684---tone-1-1 Button2465625684---variant-1-1 Button2465625684---vol-1-3 for-which-nobody-can-deny"
  data-testid="button-data-testid"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <span
    className="Button2465625684__inner"
  >
    I love big buttons and I cannot lie...
  </span>
</button>
`;

exports[`Button renders as third party (react-router) anchor provided via #as and #to props 1`] = `
<a
  className="Button2465625684__root Button2465625684---tone-1-1 Button2465625684---variant-1-1 Button2465625684---vol-1-3"
  data-testid="button-data-testid"
  href="/internal-link"
  onClick={[Function]}
>
  <span
    className="Button2465625684__inner"
  >
    I love big buttons and I cannot lie...
  </span>
</a>
`;

exports[`Button renders with stylable state prop classes, #tone, #variant, #vol and #fullwidth 1`] = `
<button
  className="Button2465625684__root Button2465625684--fullWidth Button2465625684---tone-1-6 Button2465625684---variant-1-6 Button2465625684---vol-1-6"
  data-testid="button-data-testid"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <span
    className="Button2465625684__inner"
  >
    I love big buttons and I cannot lie...
  </span>
</button>
`;
